# 타이타닉 생존 분석: 탐색적 데이터 분석 (EDA)

## 1. 개요
이 노트북에서는 전처리된 타이타닉 데이터셋을 사용하여 탐색적 데이터 분석을 수행합니다.

### 1.1 분석 목표
- 승객 구성의 전반적인 특성 파악
- 생존에 영향을 미친 주요 요인 탐색
- 사회경제적 요인과 가족 관계의 상호작용 분석

### 1.2 필요한 라이브러리 및 데이터 로드

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# 시각화 설정
plt.style.use('classic')
sns.set_theme()
%matplotlib inline

# 데이터 로드
df = pd.read_csv('../data/processed/titanic_processed.csv')

# 기본 정보 확인
print("데이터셋 기본 정보:")
print(df.info())

# 기술 통계량 확인
print("\n기술 통계량:")
print(df.describe())

데이터셋 기본 정보:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 16 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   survived       891 non-null    int64  
 1   pclass         891 non-null    int64  
 2   sex            891 non-null    object 
 3   age            891 non-null    float64
 4   sibsp          891 non-null    int64  
 5   parch          891 non-null    int64  
 6   fare           891 non-null    float64
 7   embarked       891 non-null    object 
 8   who            891 non-null    object 
 9   adult_male     891 non-null    bool   
 10  deck           891 non-null    object 
 11  alone          891 non-null    bool   
 12  family_size    891 non-null    int64  
 13  family_type    891 non-null    object 
 14  age_group      891 non-null    object 
 15  fare_category  891 non-null    object 
dtypes: bool(2), float64(2), int64(5), object(7)
memory usage: 99.3+ KB
None

기술 통계량:
     

## 2. 기본 데이터 분석

### 2.1 데이터셋 특성 분석
1. 전반적인 특성:
   - 총 891명의 승객 데이터
   - 16개의 특성 (수치형, 범주형, 불리언)
   - 결측치 없음 (전처리 완료)

2. 주요 수치형 변수 특성:
   - survived: 평균 0.384 (38.4% 생존율)
   - age: 평균 29.7세 (범위: 0.42-80세)
   - fare: 평균 32.2 (범위: 0-512.3)
   - family_size: 평균 1.9명 (범위: 1-11명)

### 2.2 기초 분석 시각화
다음 시각화를 통해 주요 변수들의 분포와 관계를 살펴보겠습니다: